packit/utils/obs_helper.py

```diff
@@ -0,0 +1,454 @@
+    targets: str,
+    owner: Optional[str],
+    package_config: PackageConfig,
+    description: Optional[str],
+):
+    """
+    Creates a new OBS project and its associated package, ensuring only a single
+    package is included per call.
+
+    Args:
+        project (str, optional): The desired name for the OBS project. Defaults to None.
+        targets (str): Comma-separated list of build targets for the project. Defaults to "".
+        owner (Optional[str], optional): The owner of the project. Defaults to None.
+        package_config (PackageConfig): The config for the package to be included in the project.
+        description (Optional[str], optional): A description for the project. Defaults to None.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the created project name and associated package name.
+
+    Raises:
+        ValueError: If the provided package_config contains multiple packages.
+    """
+    conf.get_config()
+    owner = owner or conf.config["api_host_options"][_API_URL]["user"]
+    project_name = project or f"home:{owner}:packit"
+
+    targets_list = targets.split(",")
+    for target in targets_list:
+        if target in DEPRECATED_TARGET_MAP:
+            logger.warning(
+                f"Target '{target}' is deprecated. "
+                f"Please use '{DEPRECATED_TARGET_MAP[target]}' instead.",
+            )
+
+    project_metadata = targets_to_project_meta(
+        targets=targets_list,
+        owner=owner,
+        project_name=project_name,
+        description=description,
+    )
+
+    logger.info(f"Using OBS project name = {project_name}")
+
+    project_url = core.makeurl(
+        _API_URL,
+        ["source", project_name, "_meta"],
+    )
+    metafile = core.metafile(project_url, ET.tostring(project_metadata))
+    metafile.sync()
+
+    package_names = list(package_config.packages.keys())
+
+    if len(package_names) != 1:
+        raise ValueError("Cannot handle multiple packages in package_config")
+
+    create_package(project_name, (package_name := package_names[0]))
+
+    return project_name, package_name
+
+
+def init_obs_project(
+    build_dir: str,
+    package_name: str,
+    project_name: str,
+) -> Path:
+    """
+    Initializes an Open Build Service (OBS) project.
+
+    Args:
+        build_dir (str): Base directory for the project (local path).
+        package_name (str): Name of the package to be built.
+        project_name (str): Name of the OBS project to create.
+
+    Returns:
+        Path: Path to the empty package directory within the OBS project.
+    """
+    core.Project.init_project(
+        _API_URL,
+        (prj_dir := Path(build_dir)),
+        project_name,
+    )
+
+    (pkg_dir := (prj_dir / package_name)).mkdir()
+    core.checkout_package(
+        _API_URL,
+        project_name,
+        package_name,
+        prj_dir=prj_dir,
+        pathname=pkg_dir,
+    )
+
+    pkg = core.Package(pkg_dir)
+
+    for fname in os.listdir(pkg_dir):
```

Comment:
Why?

Reply:
Deletes the existing revision of the build. so it can be replaced by the current build